appservice/src/createAppService/AppInsightsCreateStep.ts

```diff
@@ -20,70 +20,89 @@ export class AppInsightsCreateStep extends AzureWizardExecuteStep<IAppServiceWiz
+                ext.outputChannel.appendLine(localize('existingNewAppInsights', 'Using existing Application Insights resource "{0}".', aiName));
             } catch (error) {
                 const pError: IParsedError = parseError(error);
                 // Only expecting a resource not found error if this is a new component
                 if (pError.errorType === 'ResourceNotFound') {
-                    const creatingNewAppInsights: string = localize('creatingNewAppInsightsInsights', 'Creating new application insights component "{0}"...', wizardContext.newSiteName);
+                    const creatingNewAppInsights: string = localize('creatingNewAppInsightsInsights', 'Creating new Application Insights resource "{0}"...', wizardContext.newSiteName);
                     ext.outputChannel.appendLine(creatingNewAppInsights);
                     progress.report({ message: creatingNewAppInsights });
 
-                    wizardContext.appInsightsComponent = await client.components.createOrUpdate(rgName, aiName, { kind: 'web', applicationType: 'web', location });
-                    const createdNewAppInsights: string = localize('createdNewAppInsights', 'Created new application insights component "{0}"...', aiName);
+                    wizardContext.appInsightsComponent = await client.components.createOrUpdate(rgName, aiName, { kind: 'web', applicationType: 'web', location: appInsightsLocation });
+                    const createdNewAppInsights: string = localize('createdNewAppInsights', 'Created new Application Insights resource "{0}"...', aiName);
                     ext.outputChannel.appendLine(createdNewAppInsights);
                 } else {
                     throw error;
                 }
             }
         } else {
-            const appInsightsNotAvailable: string = localize('appInsightsNotAvailable', 'Skipping creating an application insights component because it isn\'t compatible with this location.');
+            const appInsightsNotAvailable: string = localize('appInsightsNotAvailable', 'Skipping creating an Application Insights resource because it isn\'t compatible with this location.');
             ext.outputChannel.appendLine(appInsightsNotAvailable);
         }
     }
 
     public shouldExecute(wizardContext: IAppServiceWizardContext): boolean {
-        return !wizardContext.appInsightsComponent;
+        return !wizardContext.appInsightsComponent && !!wizardContext.newAppInsightsName;
+    }
+
+    // returns the supported location, a location in the region map, or undefined
+    private async getSupportedLocation(wizardContext: IAppServiceWizardContext, location: Location): Promise<string | undefined> {
+        // tslint:disable-next-line: strict-boolean-expressions
+        const locations: string[] = await this.getLocations(wizardContext) || [];
+        const locationName: string = nonNullProp(location, 'name');
+        // still need to do a name step
```

Comment:
What does `still need to do a name step` mean?

Reply:
😅 This was to make sure that I covered this issue https://github.com/microsoft/vscode-azuretools/issues/560